droidlet/memory/robot/loco_memory_nodes.py

```diff
@@ -46,9 +46,12 @@ def create(cls, memory, detected_obj) -> str:
             detected_obj.get_xyz()["z"],
             cls.NODE_TYPE,
         )
+        # TODO: remove the round-about serialization,
+        # and improve performance
+        struct = detected_obj.to_struct()
```

Comment:
so this is the place where the test is failing rn, I think bc there is a mock detection object defined in the test that doesn't have https://github.com/facebookresearch/fairo/blob/hello_robot/droidlet/perception/robot/handlers/detector.py#L193 
defined.   I wonder if we could use the "official" detection object somehow in the test instead of mocking it?

Reply:
i've fixed the test.
also this was a really roundabout way to serialize, which is why I added the TODO.
I've cleaned it up properly (and removed the TODO)